### DISCUSSION TITLE
How to deploy server and give access to query it only to a particular website

### DISCUSSION THREAD

**Discussion by SuPythony** [mirumee/ariadne#647] [Q&A]
Hi,

I have made a basic API using Aridane with ariadne.asgi which runs perfectly using uvicorn. I store the data in an sqlite database using python's sqlite3 library.

Now I am not able to understand how do I host the server so that it runs and also has access to the .db file. I want to deploy using heroku. Also how can I restrict it's use only to my React application.

Thank you!

**Comment by rafalp** [comment]
Running Ariadne's ASGI app directly through Uvicorn (or other ASGI server) is nice starting point for experimenting with GraphQL APIs, but for real use with databases and access control you should embed your GraphQL app in more featured ASGI framework like [Starlette](https://starlette.io) or [FastAPI](https://fastapi.tiangolo.com/).

Eg.:

https://fastapi.tiangolo.com/advanced/async-sql-databases/
https://fastapi.tiangolo.com/tutorial/middleware/
https://fastapi.tiangolo.com/deployment/

**Reply by SuPythony** [reply]
So using just uvicorn is not suitable for production?

**Reply by rafalp** [reply]
> So using just uvicorn is not suitable for production?

This is not what I am saying. I am saying that you need to include middle layer between Uvicorn and Ariadne to implement the features you need.

**Reply by SuPythony** [reply]
Ok! I have added a starlette middleware like this - 

```
app = CORSMiddleware(GraphQL(schema, debug=False, introspection=False,
                             context_value=get_context_value), allow_origins=[os.getenv("ALLOWED_ORIGIN")],
                     allow_methods=("GET", "POST", "OPTIONS"),
                     allow_headers=("authorization", "Origin", "Access-Control-Request-Method"),)
```

CORSMiddleware is imported from `starlette.middleware.cors`

Is it fine or should I add something more?
